File: examples/example-jdbc-spring/src/main/java/example/controllers/OwnerController.java
package example.controllers;

import java.util.List;
import java.util.Optional;

import javax.validation.constraints.NotBlank;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import example.domain.Owner;
import example.repositories.OwnerRepository;

@RestController
@RequestMapping("/owners")
class OwnerController {

    private final OwnerRepository ownerRepository;

    OwnerController(OwnerRepository ownerRepository) {
        this.ownerRepository = ownerRepository;
    }

    @GetMapping("/")
    List<Owner> all() {
        return ownerRepository.findAll();
    }

    @GetMapping("/{name}")
    Optional<Owner> byName(@PathVariable("name") @NotBlank String name) {
        return ownerRepository.findByName(name);
    }
}

File: examples/example-jdbc-kotlin/src/main/kotlin/example/controllers/OwnerController.kt
package example.controllers

import java.util.Optional

import javax.validation.constraints.NotBlank

import example.domain.Owner
import example.repositories.OwnerRepository
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get

@Controller("/owners")
open class OwnerController(private val ownerRepository: OwnerRepository) {

    @Get("/")
    fun all(): List<Owner> {
        return ownerRepository.findAll()
    }

    @Get("/{name}")
    open fun byName(@NotBlank name: String): Optional<Owner> {
        return ownerRepository.findByName(name)
    }
}